test/images/regression-issue-74839/main.go

```diff
@@ -117,14 +136,44 @@ func probe(ip string) {
 	}
 }
 
-func getIP() net.IP {
-	conn, err := net.Dial("udp", "8.8.8.8:53")
+// getIPs return one IPv4 and IPv6 global ip address
+func getIPs() (ipv4, ipv6 net.IP) {
```

Comment:
two reasons, 
- this is more resilient to API changes, actually we pluralised this field without a feature gate an is causing issues.
- I didn't want to modify the original code too much, just that it works for both ipv4 and ipv6.

Reply:
I disagree; the new code to figure out the correct IP is complicated, and if it ends up guessing a different IP than the one that the caller is using, then the result will be that it doesn't correctly RST the connection. It's better to just be told the canonical IP.

Reply:
>  then the result will be that it doesn't correctly RST the connection

and the test will fail if that happens.

But you are right, this will be a lottery with multiple interfaces, will change it accordingly

Reply:
done